crates/oxc_linter/src/config/config_store.rs

```diff
@@ -110,6 +129,18 @@ impl Config {
             .collect::<Vec<_>>();
 
         for override_config in overrides_to_apply {
+            if let Some(override_plugins) = override_config.plugins {
+                if override_plugins != plugins {
+                    for (rule, severity) in all_rules.iter().filter_map(|rule| {
+                        self.categories
+                            .get(&rule.category())
+                            .map(|severity| (rule.clone(), severity))
+                    }) {
+                        rules.entry(rule).or_insert(*severity);
+                    }
+                }
+            }
+
```

Comment:
before applying any rules in the override, we need to check if the plugins have changed

if the plugins have changed, we need to re-apply the rules based on the categories at the root of the config